dirty_cat/_super_vectorizer.py

```diff
@@ -427,6 +429,15 @@ def _apply_cast(self, X: pd.DataFrame) -> pd.DataFrame:
         for col in self.imputed_columns_:
             X.loc[:, col] = _replace_missing_in_cat_col(X[col])
         for col, dtype in self.types_.items():
+            # if categorical, add the new categories to prevent
+            # them to be encoded as nan
+            if pd.api.types.is_categorical_dtype(dtype):
+                known_categories = dtype.categories
+                new_categories = pd.unique(X[col])
+                dtype = pd.CategoricalDtype(
+                    categories=known_categories.union(new_categories)
+                )
+                self.types_[col] = dtype
```

Comment:
That part is really subtle (I would have gotten it wrong). Do we have tests that fail if it is not coded right?

Reply:
Good point. We have a test that checks that the output is right, but I'll add a test to check that the format is right too.

Reply:
Actually I've deleted this and converted every categorical column to `object` dtype, to avoid dealing with the categories attached to a dtype. I think it makes everything simpler, but I'm tagging @LilianBoulard to make sure this isn't wrong (I've deleted some of your code dealing with categories).